07.label_semantic_roles/README.cn.md

```diff
@@ -187,316 +187,351 @@ conll05st-release/
-ctx_n1 = paddle.layer.data(name='ctx_n1_data', type=d_type(word_dict_len))
-ctx_0 = paddle.layer.data(name='ctx_0_data', type=d_type(word_dict_len))
-ctx_p1 = paddle.layer.data(name='ctx_p1_data', type=d_type(word_dict_len))
-ctx_p2 = paddle.layer.data(name='ctx_p2_data', type=d_type(word_dict_len))
-
-# 谓词上下区域标志
-mark = paddle.layer.data(name='mark_data', type=d_type(mark_dict_len))
-
-# 标注序列
-target = paddle.layer.data(name='target', type=d_type(label_dict_len))
+mark_dict_len = 2
+word_dim = 32
+mark_dim = 5
+hidden_dim = 512
+depth = 8
+mix_hidden_lr = 1e-3
+
+IS_SPARSE = True
+PASS_NUM = 10
+BATCH_SIZE = 10
+
+embedding_name = 'emb'
 ```
 
 这里需要特别说明的是hidden_dim = 512指定了LSTM隐层向量的维度为128维，关于这一点请参考PaddlePaddle官方文档中[lstmemory](http://www.paddlepaddle.org/doc/ui/api/trainer_config_helpers/layers.html#lstmemory)的说明。
 
-- 将句子序列、谓词、谓词上下文、谓词上下文区域标记通过词表，转换为实向量表示的词向量序列。
+- 如上文提到，我们用基于英文维基百科训练好的词向量来初始化序列输入、谓词上下文总共6个特征的embedding层参数，在训练中不更新。
 
 ```python  
-
-# 在本教程中，我们加载了预训练的词向量，这里设置了：is_static=True
-# is_static 为 True 时保证了在训练 SRL 模型过程中，词表不再更新
-emb_para = paddle.attr.Param(name='emb', initial_std=0., is_static=True)
-# 设置超参数
-default_std = 1 / math.sqrt(hidden_dim) / 3.0
-std_default = paddle.attr.Param(initial_std=default_std)
-std_0 = paddle.attr.Param(initial_std=0.)
-
-predicate_embedding = paddle.layer.embedding(
-    size=word_dim,
-    input=predicate,
-    param_attr=paddle.attr.Param(
-        name='vemb', initial_std=default_std))
-mark_embedding = paddle.layer.embedding(
-    size=mark_dim, input=mark, param_attr=std_0)
-
-word_input = [word, ctx_n2, ctx_n1, ctx_0, ctx_p1, ctx_p2]
-emb_layers = [
-    paddle.layer.embedding(
-        size=word_dim, input=x, param_attr=emb_para) for x in word_input
-]
-emb_layers.append(predicate_embedding)
-emb_layers.append(mark_embedding)
+def load_parameter(file_name, h, w):
+    with open(file_name, 'rb') as f:
+        f.read(16)  # skip header.
+        return np.fromfile(f, dtype=np.float32).reshape(h, w)
 ```
 
 - 8个LSTM单元以“正向/反向”的顺序对所有输入序列进行学习。
 
 ```python  
-hidden_0 = paddle.layer.mixed(
-    size=hidden_dim,
-    bias_attr=std_default,
-    input=[
-        paddle.layer.full_matrix_projection(
-            input=emb, param_attr=std_default) for emb in emb_layers
-    ])
-
-mix_hidden_lr = 1e-3
-lstm_para_attr = paddle.attr.Param(initial_std=0.0, learning_rate=1.0)
-hidden_para_attr = paddle.attr.Param(
-    initial_std=default_std, learning_rate=mix_hidden_lr)
-
-lstm_0 = paddle.layer.lstmemory(
-    input=hidden_0,
-    act=paddle.activation.Relu(),
-    gate_act=paddle.activation.Sigmoid(),
-    state_act=paddle.activation.Sigmoid(),
-    bias_attr=std_0,
-    param_attr=lstm_para_attr)
-
-#stack L-LSTM and R-LSTM with direct edges
-input_tmp = [hidden_0, lstm_0]
-
-for i in range(1, depth):
-    mix_hidden = paddle.layer.mixed(
+def db_lstm(word, predicate, ctx_n2, ctx_n1, ctx_0, ctx_p1, ctx_p2, mark,
+            **ignored):
```

Comment:
What does the `**ignored` do? It is not used anywhere.

Reply:
Done